CommunityFridgeMapApi/dependencies/python/db.py

```diff
@@ -51,13 +70,57 @@ class DB_Item:
     REQUIRED_FIELDS = []
     ITEM_TYPES = {}
     TABLE_NAME = ""
+    FIELD_VALIDATION = {}
 
     def __init__(self, db_client: "botocore.client.DynamoDB"):
         self.db_client = db_client
         pass
 
-    def add_item(self):
-        pass
+    def add_item(self, conditional_expression=None) -> DB_Response:
+        """
+        adds item to database
+            Parameters:
+                conditional_expression (str): conditional expression for boto3 function put_item
+
+            Returns:
+                db_response (DB_Response): returns a DB_Response
+        """
+        field_validation = self.validate_fields()
+        if not field_validation.is_valid():
+            return DB_Response(
+                message=field_validation.get_message(), status_code=400, success=False
+            )
+        item = self.format_dynamodb_item_v2()
+        try:
+            self.db_client.put_item(
+                TableName=self.TABLE_NAME,
+                Item=item,
+                ConditionExpression=conditional_expression,
+            )
+        except self.db_client.exceptions.ConditionalCheckFailedException as e:
+            return DB_Response(
+                message=f"{self.TABLE_NAME} already exists, pick a different Name",
+                status_code=409,
+                success=False,
+            )
+        except self.db_client.exceptions.ResourceNotFoundException as e:
+            message = (
+                "Cannot do operations on a non-existent table:  %s" % self.TABLE_NAME
```

Comment:
I see `f-strings` everywhere else, why not here as well?

Reply:
there's a mix of f-strings and this formatting in the tag class I did as well and I think a few other places as well.  I don't know if there's been a determination on which formatting to follow.

Reply:
lets stick with f-strings to be consistent

Reply:
done: https://github.com/CollectiveFocus/CFM_Backend/pull/11/commits/524190fd14bd6ae3b0d578437e2484b3a0231b4f